lib/js/components/Switch/Switch.vue

```diff
@@ -0,0 +1,350 @@
+@import '../../../styles/settings/colors/tokens';
+@import '../../../styles/settings/typography/tokens';
+
+$switch-transition: all $default-transition-time ease-out;
+
+.switchWrapper {
+	display: inline-flex;
+	max-width: 100%;
+	overflow: hidden;
+}
+
+.switch {
+	$root: &;
+
+	align-items: flex-start;
+	background-color: $color-default-background;
+	border: 1px solid $color-neutral-border;
+	border-radius: $radius-xs;
+	box-sizing: border-box;
+	display: inline-flex;
+	max-width: 100%;
+	position: relative;
+
+	&.-rounded {
+		border-radius: $radius-xl;
+	}
+
+	&__icon {
+		color: $color-neutral-icon;
+	}
+
+	.-selected &__icon {
+		color: $color-primary-icon;
+	}
+
+	&__label {
+		@include label-s-extensive-bold-uppercase;
+
+		color: $color-neutral-text;
+		overflow: hidden;
+		text-overflow: ellipsis;
+		white-space: nowrap;
+	}
+
+	.-selected &__label {
+		color: $color-neutral-text-heavy;
+	}
+
+	&__item {
+		align-items: center;
+		background-color: $color-neutral-background-ghost;
+		border: 1px solid transparent;
+		border-radius: $radius-xs;
+		cursor: pointer;
+		display: flex;
+		flex-grow: 1;
+		gap: $space-xxxs;
+		justify-content: center;
+		margin: -1px;
+		min-height: 32px;
+		min-width: 64px;
+		overflow: hidden;
+		padding: 0 $space-s;
+		position: relative;
+		transition: $switch-transition;
+		z-index: 2;
+
+		&.-left {
+			margin: -1px 0 -1px -1px;
+		}
+
+		&.right {
+			margin: -1px -1px -1px 0;
+		}
+
+		&.-selection {
+			background-color: $color-primary-background;
+			border: 1px solid $color-primary-border;
+			box-sizing: border-box;
+			left: 0;
+			position: absolute;
+			top: 0;
+			z-index: 1;
+
+			#{$root}__icon {
+				visibility: hidden;
+			}
+
+			#{$root}__label {
+				overflow: hidden;
+				visibility: hidden;
+				white-space: nowrap;
+				width: max-content;
+			}
+
+			&.-left {
+				left: 0;
+			}
+
+			&.-right {
+				left: calc(100% + 2px);
+				transform: translateX(-100%);
+			}
+		}
+
+		&.-selected,
+		&.-selection:hover {
+			cursor: default;
+			flex-grow: 0.00001;
+			flex-shrink: 0;
+			overflow: hidden;
+			text-overflow: initial;
+		}
+	}
+
+	&.-small &__item {
+		gap: $space-xxxxs;
+		min-height: 24px;
+		min-width: 48px;
+		padding: 0 $space-xs;
+	}
+
+	&.-rounded &__item {
+		border-radius: $radius-xl;
+	}
+
+	&.-disabled &__item {
+		cursor: default;
+
+		&.-selected {
+			background-color: $color-primary-background-disabled;
+			border-color: $color-primary-border-disabled;
+		}
+	}
+
+	&.-disabled &__icon {
+		color: $color-neutral-icon-disabled;
+	}
+
+	&.-disabled .-selected &__icon {
+		color: $color-primary-icon-disabled;
+	}
+
+	&.-disabled &__label {
+		color: $color-neutral-text-weak;
+	}
+
+	&.-disabled .-selected &__label {
+		color: $color-neutral-text-strong;
+	}
+
+	&:not(.-disabled) &__item:not(.-selected) {
+		&:hover {
+			background-color: $color-neutral-background-ghost-hovered;
+
+			&::before {
+				background-color: $color-neutral-ripple;
```

Comment:
I think ripple effect should be shared across all DS components, but I need to clarify it first.
https://bethinkteam.slack.com/archives/C018TDAH5FW/p1710777401086249